aws/internal/service/synthetics/waiter/status.go

```diff
@@ -0,0 +1,29 @@
+package waiter
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/synthetics"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func CanaryStatus(conn *synthetics.Synthetics, name string) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		input := &synthetics.GetCanaryInput{
+			Name: aws.String(name),
+		}
+
+		output, err := conn.GetCanary(input)
```

Comment:
Add `CabaryByName` to a `finder` package and use here, resource Read and acceptance tests.